diff -r -U 3 minecraft\net\minecraft\client\renderer\texture\TextureMap.java minecraft_patched\net\minecraft\client\renderer\texture\TextureMap.java
--- minecraft\net\minecraft\client\renderer\texture\TextureMap.java
+++ minecraft_patched\net\minecraft\client\renderer\texture\TextureMap.java
@@ -570,7 +570,7 @@
    private int detectMinimumSpriteSize(Map p_detectMinimumSpriteSize_1_, IResourceManager p_detectMinimumSpriteSize_2_, int p_detectMinimumSpriteSize_3_) {
       Map map = new HashMap();
 
-      for(Entry entry : p_detectMinimumSpriteSize_1_.entrySet()) {
+      for(Entry entry : (Set<Entry>)p_detectMinimumSpriteSize_1_.entrySet()) {
          TextureAtlasSprite textureatlassprite = (TextureAtlasSprite)entry.getValue();
          ResourceLocation resourcelocation = new ResourceLocation(textureatlassprite.func_94215_i());
          ResourceLocation resourcelocation1 = this.completeResourceLocation(resourcelocation);                                                                                        ;
