diff -r -U 3 minecraft\net\minecraft\client\renderer\chunk\ChunkRenderWorker.java minecraft_patched\net\minecraft\client\renderer\chunk\ChunkRenderWorker.java
--- minecraft\net\minecraft\client\renderer\chunk\ChunkRenderWorker.java
+++ minecraft_patched\net\minecraft\client\renderer\chunk\ChunkRenderWorker.java
@@ -82,7 +82,7 @@
       }
 
       Entity entity1 = Minecraft.func_71410_x().func_175606_aa();
-      if (entity == null) {
+      if (entity1 == null) {
          p_178474_1_.func_178542_e();
       } else {
          p_178474_1_.func_178541_a(this.func_178475_b());
