diff -r -U 3 minecraft\shadersmod\client\Shaders.java minecraft_patched\shadersmod\client\Shaders.java
--- minecraft\shadersmod\client\Shaders.java
+++ minecraft_patched\shadersmod\client\Shaders.java
@@ -761,7 +761,7 @@
 
    private static CustomTexture[] loadCustomTextures(Properties props, int stage) {
       String s = "texture." + STAGE_NAMES[stage] + ".";
-      Set set = props.keySet();
+      Set<String> set = (Set<String>)((Set)props.keySet());
       List<CustomTexture> list = new ArrayList<CustomTexture>();
 
       for(String s1 : set) {
@@ -991,7 +991,7 @@
             for(int i = 0; i < ashaderoption1.length; ++i) {
                ShaderOption shaderoption = ashaderoption1[i];
                if (shaderoption == null) {
-                  list.add((Object)null);
+                  list.add(null);
                } else if (shaderoption instanceof ShaderOptionRest) {
                   ShaderOption[] ashaderoption2 = getShaderOptionsRest(shaderPackGuiScreens, ashaderoption);
                   list.addAll(Arrays.asList(ashaderoption2));
@@ -1275,7 +1275,7 @@
 
    public static void startup(Minecraft mc) {
       checkShadersModInstalled();
-      mc = mc;
+      Shaders.mc = mc;
       mc = Minecraft.func_71410_x();
       capabilities = GLContext.getCapabilities();
       glVersionString = GL11.glGetString(7938);
@@ -1785,7 +1785,7 @@
    }
 
    private static int createFragShader(String filename) {
-      int i = ARBShaderObjects.glCreateShaderObjectARB(35632);
+      int i = ARBShaderObjects.glCreateShaderObjectARB(org.lwjgl.opengl.ARBFragmentShader.GL_FRAGMENT_SHADER_ARB);
       if (i == 0) {
          return 0;
       } else {
@@ -2956,11 +2956,11 @@
 
    private static void checkWorldChanged(World world) {
       if (currentWorld != world) {
-         World world = currentWorld;
+         World prevWorld = currentWorld;
          currentWorld = world;
-         if (world != null && world != null) {
-            int i = world.field_73011_w.func_186058_p().func_186068_a();
-            int j = world.field_73011_w.func_186058_p().func_186068_a();
+         if (prevWorld != null && currentWorld != null) {
+            int i = prevWorld.field_73011_w.func_186058_p().func_186068_a();
+            int j = currentWorld.field_73011_w.func_186058_p().func_186068_a();
             boolean flag = shaderPackDimensions.contains(Integer.valueOf(i));
             boolean flag1 = shaderPackDimensions.contains(Integer.valueOf(j));
             if (flag || flag1) {
@@ -3804,7 +3804,7 @@
 
    public static void setFog(GlStateManager.FogMode fogMode) {
       GlStateManager.func_187430_a(fogMode);
-      fogMode = fogMode.field_187351_d;
+      Shaders.fogMode = fogMode.field_187351_d;
       if (fogEnabled) {
          setProgramUniform1i("fogMode", fogMode.field_187351_d);
       }
@@ -3941,7 +3941,7 @@
                   Lang.loadLocaleData(inputstream, properties);
                   inputstream.close();
 
-                  for(String s4 : properties.keySet()) {
+                  for(String s4 : (Set<String>)((Set)properties.keySet())) {
                      String s5 = properties.getProperty(s4);
                      shaderPackResources.put(s4, s5);
                   }
