diff -r -U 3 minecraft\shadersmod\client\ShadersTex.java minecraft_patched\shadersmod\client\ShadersTex.java
--- minecraft\shadersmod\client\ShadersTex.java
+++ minecraft_patched\shadersmod\client\ShadersTex.java
@@ -735,7 +735,7 @@
       int k = 0;
       int[] aint = null;
 
-      for(String s : list) {
+      for(String s : (List<String>)list) {
          if (s != null) {
             try {
                ResourceLocation resourcelocation = new ResourceLocation(s);
